code/game/machinery/newscaster.dm

```diff
@@ -132,18 +133,16 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 	name = "Security Newscaster"
 	securityCaster = 1
 
-/obj/machinery/newscaster/New(turf/loc, var/ndir=0)
-	if(ndir)
-		dir=ndir
-	else
-		dir = reverse_direction(dir) //Unfucks all maps
-	pixel_x = (dir & 3)? 0 : (dir == 4 ? 28 : -28)
-	pixel_y = (dir & 3)? (dir ==1 ? 33 : -28) : 0
+/obj/machinery/newscaster/New(var/loc, var/ndir, var/building = 1)
+	buildstage = building
+	if(!buildstage) //Already placed newscasters via mapping will not be affected by this
+		pixel_x = (ndir & 3)? 0 : (ndir == 4 ? 28 : -28)
+		pixel_y = (ndir & 3)? (ndir == 1 ? 28 : -28) : 0
+		dir = ndir
```

Comment:
I do want to note that lines 138-141 right here were all that were needed for the original fix, and it wouldn't have messed with pre-mapped ones. I stole this from light switch code.